src/core/load-fragment.c

```diff
@@ -3913,9 +3936,14 @@ int config_parse_memory_limit(
                 c->memory_max = bytes;
         else if (streq(lvalue, "MemorySwapMax"))
                 c->memory_swap_max = bytes;
-        else if (streq(lvalue, "MemoryLimit"))
+        else if (streq(lvalue, "MemoryLimit")) {
+
+                log_syntax(unit, LOG_WARNING, filename, line, 0,
+                           "Unit configured to use MemoryLimit=. "
+                           "This is an obsolete cgroupv1 setting, please update to use MemoryMax= instead. Support for MemoryLimit= will be removed soon.");
```

Comment:
"Unit uses MemoryLimit=; please use MemoryMax= instead. Support for MemoryLimit= will be removed soon."